### ISSUE TITLE
[FEATURE REQUEST] Option to install third party stub packages to be used with the embedded mypy interpreter

### ISSUE THREAD

**Issue by vancromy** [microsoft/vscode-mypy#102]
Hello,

Apologies if this has been answered already. I tried a quick search of issues and PRs but got no hits. How is this extension suggesting to handle third-party stub packages? I am for example getting a `Library stubs not installed for "yaml"` error in many of my files which clogs up my problems window. I am aware that I can adjust the settings to not display these types of errors but I would like to use third-party stub packages if they are available. Installing `types-PyYAML` in my local .venv does not work since the extension is using the packaged mypy interpreter. I could also tell the extension to use a custom mypy interpreter from my .venv with the third-party packages installed but it would be a nice feature to be able to embed these with the packaged mypy interpreter so that I could use the same setup across several projects.

**Comment by karthiknadig** [comment]
@vancromy Can you try the following and see if it works for you. 
Steps from a terminal:
1. Create a directory in some location say `~/mytypes` (mac/linux) or `%USERPROFILE%/mytypes` (windows)
2. Use `python -m pip install -t ~/mytypes  <packages>` (mac/linux) or `python -m pip install -t %USERPROFILE%/mytypes  <packages>`  (windows)
3. Set environment variable `export MYPYPATH=~/mytypes` (mac/linux) or `set MYPYPATH=%USERPROFILE%/mytypes` (windows)
4. Navigate to your project directory: cd <project dir>
5. launch vscode `code .`

This will allow mypy to find the packages. Let me know if that works.

If the above works, all you need to do is set MYPYPATH=~/mytypes globally. that way you don;t have to set it from terminal each time.

**Comment by vancromy** [comment]
Hi @karthiknadig,

Thanks for the quick response. I tried implementing your steps this morning but it didn't seem to work even after Quitting vscode and starting it again. See below screenshot:
![image](https://github.com/microsoft/vscode-mypy/assets/59556820/7a7e22f8-9344-41fc-83ea-87c14b7b79d5)

I'm on macOS 13.4.1(c) and using the Pre-Releave Version of the extension.

I also wondered which python version I should use when pip installing and how this would handle projects with varying python versions. I'm assuming the types should be installed with the same python version as the bundled mypy?

**Comment by karthiknadig** [comment]
It worked for me, turns out you have to use PYTHONPATH instead of MYPYPATH (sorry about that):
![image](https://github.com/microsoft/vscode-mypy/assets/3840081/febe5406-d2ea-4320-83a6-622f580a47dd)

![image](https://github.com/microsoft/vscode-mypy/assets/3840081/b3696f61-9348-4e91-852f-7e22a3a1a79c)

**Comment by vancromy** [comment]
Hi @karthiknadig,

Thanks for the update. I haven't had the chance to test this yet but it also seems like mypy is not complaining about me missing type stub packages (v2023.3.12191006). Did you guys change something since I opened this issue (e.g. turn off 3-party stub package warnings by default)?

**Comment by vancromy** [comment]
Ah I see #105 explains why I no longer see anything. Ok I used the workaround from #105 and also did the following based on the link you shared:

1. Create a virtual environment to host all my stub packages
2. Install the necessary packages in that venv
3. Update the extension settings to add a new arg for `--python-executable=path/to/type/stubs/venv` ([--python-execuable](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-python-executable))

Now I don't get complaints due to missing 3rd party stub packages. What I haven't found out is how mypy handles different python versions? I did not try changing PYTHONPATH as you suggested because I was unsure about the implications this would have for all my other python envs.
